src/day8/re_frame/trace.cljs

```diff
@@ -144,20 +144,27 @@
     (fn [trace]
       (< (:query query) (:duration trace)))))
 
-(defn render-traces [showing-traces filter-items trace-detail-expansions]
+(defn add-filter [filter-items filter-input filter-type from-mouse?]
+  (swap! filter-items conj {:id (random-uuid)
+                            :query (if from-mouse?
+                                     (str/lower-case (name filter-input))
+                                     (if (= @filter-type :contains)
+                                       (str/lower-case @filter-input)
```

Comment:
Like with `filter-type`, it's probably a bit nicer if we deref `filter-input` in the call to the `add-filter` rather than inside it.